aten/src/ATen/native/TensorShape.cpp

```diff
@@ -2456,4 +2456,21 @@ std::vector<Tensor> unflatten_dense_tensors(const Tensor& flat, TensorList tenso
   return outputs;
 }
 
-}} // at::native
+at::Tensor slice_scatter(const at::Tensor& base, const at::Tensor& mutated_view, int64_t dim, c10::optional<int64_t> start, c10::optional<int64_t> end, int64_t step) {
+    auto output = base.clone();
+    output.slice(dim, start, end, step).copy_(mutated_view);
+    return output;
+}
+at::Tensor select_scatter(const at::Tensor& base, const at::Tensor& mutated_view, int64_t dim, int64_t index) {
+    auto output = base.clone();
+    output.select(dim, index).copy_(mutated_view);
+    return output;
+}
+at::Tensor diagonal_scatter(const at::Tensor& base, const at::Tensor& mutated_view, int64_t offset, int64_t dim1, int64_t dim2) {
+    auto output = base.clone();
+    output.diagonal(offset, dim1, dim2).copy_(mutated_view);
+    return output;
+}
```

Comment:
Are we exposing these to python? (We probably should, if users want to write code without mutations). If so we should add docstrings for these

Reply:
This PR currently will expose them to python. You're right, I should add doc strings. Is the right place for them in `torch/_torch_docs.py`? (I'll put them near their corresponding view docstrings)

Reply:
Yes, we want entries in `torch/_torch_docs.py`, and also entries in `torch/_tensor_docs.py` that link to the ones in `torch/_torch_docs.py`

For example, [tensor.abs documentation](https://pytorch.org/docs/stable/generated/torch.Tensor.abs.html?highlight=tensor%20abs#torch.Tensor.abs) links to [torch.abs docs](https://pytorch.org/docs/stable/generated/torch.abs.html#torch.abs)

Reply:
Thanks. Although, I see that `slice` is currently not exposed to python (it's hardcoded in a [list in the codegen](https://github.com/pytorch/pytorch/blob/2853c7da2243611c6fbd8b78b93c0cd34f0f1bc1/tools/autograd/gen_python_functions.py#L85)). I'm thinking of doing the same for `slice_scatter` to avoid confusion.

Reply:
Aha. The reason why slice is not exposed to python is that it's literally just indexing with slices (e.g. `x[0:2]`). We might still want to expose a slice_scatter though for people who want to write code without mutations

Reply:
Will do.

I also see that diagonal has both function/method docs, but slice is missing function docs; I figure I'll add them in for consistent (basically copying the method doc into the functions file, and adding a reference from [method docs] -> [function docs], which is what `diagonal` currently does)